### PR TITLE
SCT Feedback tweaking.

### PR DESCRIPTION
**Pull request by tomrittervg**
1) Move the format ahead of the Client/Server/Auditor operations
2) Ammend the format to what we discussed
3) Split the Client operation from the server operation
4) Move the talk about local trust anchors to the client operation section
5) Talk about the two possible server configuration modes in the Server Op Section
6) Talk about how the auditor may get the data in one of two ways in the Server to Auditor section

### CODE REVIEW

File: draft-ietf-trans-gossip.md [hunk 1]
```diff
@@ -268,15 +302,22 @@ learn about other sites visited by the HTTPS client. Second, auditors
...
-send to the server SCTs in the store that are associated with that
-domain name unless they were received from that server in the current TLS session.
-
-\[TODO: fix the above paragraph -- it is vague and confusing. maybe
-  an example including a client caching at most one SCT per host+log
-  would clarify\]
+When the client later connects to the HTTPS server it again receives a 
+set of SCTs and calculates a certificate chain, and again creates a 
+sct_feedback or similar object. If this object does not exactly match 
+an existing object in the store, then the client MUST add this new
```
Comment (ln5) [inline draft-ietf-trans-gossip.md:308]: Does this MUST contradict the SHOULD on line 285?

File: draft-ietf-trans-gossip.md [hunk 2]
```diff
@@ -308,44 +343,116 @@ data SHOULD be sent in an already established TLS session. This makes
...
+If the HTTPS client has configuration options for not sending cookies
+to third parties, SCTs of third parties MUST be treated as cookies
+with respect to this setting. This prevents third party tracking
+through the use of SCTs/certificates, which would bypass the cookie
+policy.
 
-  1. if the SCT can't be verified to be a valid SCT for the
-  accompanying leaf cert, issued by a known log, the SCT SHOULD be
-  discarded
+\[ TBD: We're thinking about reversing this decision \]
```
Comment (ln5) [inline draft-ietf-trans-gossip.md:381]: Are you referring to the cookie analogy?

File: draft-ietf-trans-gossip.md [hunk 3]
```diff
@@ -308,44 +343,116 @@ data SHOULD be sent in an already established TLS session. This makes
...
+### HTTPS server operation {#feedback-srvop}
+
+HTTPS servers can be configured (or omit configuration), resulting
+in, broadly, two modes of operation. In the simpler mode, the server
+will only track leaf certificates and SCTs applicable to those 
+leaf certificates. In the more complex mode, the server will confirm 
+the client's path validation and store the certificate path. The 
+latter mode requires more configuration, but is necessary to prevent DoS
+attacks on the server's storage space.  
+
```
Comment (ln5) [inline draft-ietf-trans-gossip.md:392]: Should we add a note on why simply limiting the size of the data store opens up for a flooding/flushing attack?

**Comment by ln5** [comment]
Please merge my tjr-new-sct-format-and-stuff into your branch if you agree with those changes.

File: draft-ietf-trans-gossip.md (hunk 1 above)
Comment (tomrittervg) [inline draft-ietf-trans-gossip.md:308]: No... Line 285 is about discarding SCTs that aren't signed by a known log, and this MUST (line 308) is about adding the constructed object to the store...

File: draft-ietf-trans-gossip.md (hunk 2 above)
Comment (tomrittervg) [inline draft-ietf-trans-gossip.md:381]: Yes.

File: draft-ietf-trans-gossip.md (hunk 3 above)
Comment (tomrittervg) [inline draft-ietf-trans-gossip.md:392]: Yea.  Flooding attacks should get their own treatment down in the policy recommendations, and a sentence here.  Truth.  I need to do that.

**Comment by tomrittervg** [comment]
merged them in!

File: draft-ietf-trans-gossip.md (hunk 1 above)
Comment (ln5) [inline draft-ietf-trans-gossip.md:308]: Oh, I misread. Thanks.

File: draft-ietf-trans-gossip.md (hunk 2 above)
Comment (ln5) [inline draft-ietf-trans-gossip.md:381]: What's our current thinking wrt this? Can't recall that discussion.

**Comment by ln5** [comment]
Rebased on master, committed and pushed.

Thanks!

File: draft-ietf-trans-gossip.md (hunk 2 above)
Comment (tomrittervg) [inline draft-ietf-trans-gossip.md:381]: We haven't discussed it yet, look for the email thread "CT Gossip and Third Party Cookies"

**Pull request closed without merging**